pymeasure/instruments/instrument.py

```diff
@@ -599,3 +647,73 @@ def check_errors(self):
             return errors
         else:
             raise NotImplementedError("Non SCPI instruments require implementation in subclasses")
+
+
+class Channel(ChannelBase):
+    """The base class for channel definitions.
+
+    This class supports dynamic properties like :class:`Instrument`,
+    but requires an instrument as a parent for communication.
+    The default implementation of :meth:`write` uses :code:`str.format` to
+    exchange :code:`'{ch}'` for the channel name.
+
+    :param instrument: The instrument (an instance of :class:`~pymeasure.instruments.Instrument`)
+        to which the channel belongs.
+    :param name: Name of the channel, as it is used for the communication.
+    """
+
+    def __init__(self, parent, name):
+        super().__init__()
+        self.parent = parent
+        self.name = name
+
+    # Calls to the instrument
+    def write(self, command, **kwargs):
+        """Write a string command to the instrument.
+
+        The channel name is inserted into the command and `write_termination` appended.
+
+        :param command: command string to be sent to the instrument.
+            '{ch}' is replaced by the channel name.
+        :param kwargs: Keyword arguments for the adapter.
+        """
+        self.parent.write(command.format(ch=self.name), **kwargs)
```

Comment:
> is that channel prefix SCPI-standardised?
SCPI-99 Section 8.3.2 is about channel lists, but there might be also other references inside this document dealing with channels.

Reply:
> If your device adds some string in front of every command, you can subclass write:

Ah, of course! That's actually just what you do now. Sorry for the confusion. 

> We could make write just calling the parent and then the documentation shows the two approaches of subclassing write ("{id}" in the individual commands or "{self.id}" in the write method).

Agreed, let's do it like that! The overrides will be short, explicit, and living with the channel implementations, so it will be easy to reason about the command tweaking sequence. If we present good default solutions for the most frequent scenarios, users can easily pick it up, too.

Reply:
> So, quite a number of instruments account for channels with a common prefix to all commands, like e.g. `CH1:VOLT?`. Btw, @msmttchr is that channel prefix SCPI-standardised? Do you know?
> 


No, sorry I don't

Reply:
i think this can be resolved.